### DISCUSSION TITLE
Parsing large XML in chunks

### DISCUSSION THREAD

**Discussion by Zireael** [NaturalIntelligence/fast-xml-parser#313] [General]
Hi, I am wondering how to best go on about parsing a very large XML. For example coming as a stream from fs. This is for files that are larger than RAM available (for example running in a docker or kubernetes container, or on Raspberry Pi).

I see in examples on the main page:
```
// Intermediate obj
var tObj = parser.getTraversalObj(xmlData,options);
var jsonObj = parser.convertToJson(tObj,options);
```
But not sure if this is supposed to be used with something like a generator invoking `next()` or in a stream?

```
const { pipeline } = require('stream');
const fs = require('fs');
const parser = require("fast-xml-parser");

const myoptions = {textNodeName: "#text"}

parser.options(myoptions)

pipeline(
  fs.createReadStream('myfile.xml'),
  parser.convertToJson(),
  fs.createWriteStream('myoutput.json'),
  (err) => {
    if (err) {
      console.error('Pipeline failed', err);
    } else {
      console.log('Pipeline succeeded');
    }
  }
);
```

**Comment by amitguptagwl** [comment]
Please use https://github.com/nodable/flexible-xml-parser with Sequential Output Builder or Feedbale input source.
